src/rpc/webrtc.rs

```diff
@@ -230,19 +230,22 @@ pub(crate) async fn new_peer_connection_for_client(
     let data_channel = peer_connection
         .create_data_channel("data", Some(data_channel_init))
         .await?;
-
     let negotiation_channel = peer_connection
         .create_data_channel("negotiation", Some(negotiation_channel_init))
         .await?;
 
     let nc = negotiation_channel.clone();
-    let pc = peer_connection.clone();
+    let pc = Arc::downgrade(&peer_connection);
 
     negotiation_channel
         .on_message(Box::new(move |msg: DataChannelMessage| {
-            let pc = pc.clone();
+            let wpc = pc.clone();
             let nc = nc.clone();
             Box::pin(async move {
+                let pc = match wpc.upgrade() {
+                    Some(pc) => pc,
+                    None => return,
```

Comment:
This feels like an idiom for which there is some shorthand, but I'm not quite sure what it would be. Maybe
```rust
let pc = wpc.upgrade()?;
```
or something with `.unwrap()` or `if let`. I can't quite see it right now, but this feels like it should be 1 line instead of 3.

Reply:
Yeah, some shorthand would be nice! We can't use the `?` operator though because the return type is `()`. I don't love adding an `unwrap()` if we can avoid it, and (personal opinion only) I think wrapping the following code block in the `if let` is a bit uglier.